### PR TITLE
Enable l1-builtin-info test

### PR DESCRIPTION
**Pull request by Frassle**
This enables the l1-builtin-info test. As we're now running multiple templates with the same `yamlLanguageHost` we can't globally cache the template in loadTemplate anymore.

Instead we cache by the program directory. For normal runs there will only be the one program directory, and for conformance test we know each program will be in its own unique directory.

### CODE REVIEW

**Review by lunaris — APPROVED** [review]
I'm minded for us to spend a bit of time trying to work out why directory caching doesn't work here, since it seems like the fundamental unit that we should be able to capture. Failing that, can we push down to the load and do it at the file level?

**Comment by Frassle** [comment]
> I'm minded for us to spend a bit of time trying to work out why directory caching doesn't work here

Yeh I'll have a fresh look today. Maybe sleeping on it will have helped.

**Comment by Frassle** [comment]
Found the problem with caching, it was only if "compiler" option was set. Fixed to use directory caches and added a comment as to why we don't cache if compiler is set.

File: pkg/server/server.go [hunk 1]
```diff
@@ -84,10 +93,15 @@ func (host *yamlLanguageHost) loadTemplate(directory string, compilerEnv []strin
 	if diags.HasErrors() {
 		return nil, diags, nil
 	}
-	host.template = template
-	host.diags = diags
 
-	return host.template, diags, nil
+	if host.compiler != "" {
```
Comment (lunaris) [inline pkg/server/server.go:97]: Might be worth either adding a `shouldCache` method or even just a `cache` method that internalise this check/provide a single place we can document it.

**Pull request merged**